app/controllers/departments_controller.ts

```diff
@@ -0,0 +1,29 @@
+import type { HttpContext } from "@adonisjs/core/http";
+
+import Department from "#models/department";
+import { showValidator } from "#validators/show";
+
+export default class DepartmentsController {
+  async index({ request }: HttpContext) {
+    const departments = await Department.query().withScopes((scopes) => {
+      scopes.handleSearchQuery(request.only(["id", "createdAt", "updatedAt"]));
+      scopes.includeRelations(request.only(["departments"]));
+      scopes.handleSortQuery(request.input("sort"));
```

Comment:
- I think this endpoint should support more search query parameters than just the ID and dates. Ideally you'd only give the scope the few fields to ignore (such as the gradient colors) instead of a list of fields to allow
- `departments` is not a valid relation defined on the model. Look at the model definition to see what strings you should pass there - these should be the names of relations you defined there.
  - while at it you might as well move the list of valid relations into a readonly list on the class and reuse it in the `show` method without duplicating code